core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -465,7 +465,7 @@ private[spark] class TaskSchedulerImpl(
     var reason: Option[ExecutorLossReason] = None
     synchronized {
       try {
-        taskIdToTaskSetManager.get(tid) match {
+        Option(taskIdToTaskSetManager.get(tid)) match {
```

Comment:
Good catch. I like this direction.

I have a question about the change of semantics. By removing `synchronization` at `accumUpdatesWithTaskIds()`, a pair of operations in this `synchronized` `get()` and `remove()` in `cleanupTaskState()` is not atomic regarding `get`.
Is this change ok?

Reply:
It think your question is about the get() and remove() inside cleanupTaskState all being inside a single synchronize block, correct?  

I don't see that as being a problem here since taskIdToTaskSetManager is a concurrentHashMap.  That protects the operations from being atomic and if you do a remove on an object that isn't there then it does nothing.  There is no other code that removes from there so I don't think that can happen anyway.  With this change the only thing outside of a synchronize block is a get in accumUpdatesWithTaskIds which will be harmless if it had been removed.

Reply:
`ConcurrentHashMap` make each operation like `get()`, `remove()`, and others. Thus, I reviewed places more than one operations are within a `synchronized`. The place is here. 
When we apply this PR, the `get` in `accumUpdatesWithTaskIds` can be executed between `get()` and `remove()`. My question is like a confirmation whether it is safe or not.

Reply:
yes as far as I can see its safe.  If the get happened before its removed it calculates the accumulators, if its after its removed it just gets an empty array back.  This isn't any different then when it was synchronized.   There is nothing in the statusUpdate between the get and call to cleanupTaskState where it removes that I see depends on accumulators or anything else.